vllm/model_executor/layers/quantization/__init__.py

```diff
@@ -11,19 +11,21 @@
 from vllm.model_executor.layers.quantization.deepspeedfp import (
     DeepSpeedFPConfig)
 from vllm.model_executor.layers.quantization.fp8 import Fp8Config
+from vllm.model_executor.layers.quantization.fp8fnuz import Fp8FnuzConfig
 from vllm.model_executor.layers.quantization.gptq import GPTQConfig
 from vllm.model_executor.layers.quantization.gptq_marlin import (
     GPTQMarlinConfig)
 from vllm.model_executor.layers.quantization.gptq_marlin_24 import (
     GPTQMarlin24Config)
 from vllm.model_executor.layers.quantization.marlin import MarlinConfig
 from vllm.model_executor.layers.quantization.squeezellm import SqueezeLLMConfig
+from vllm.utils import is_hip
 
 QUANTIZATION_METHODS: Dict[str, Type[QuantizationConfig]] = {
     "aqlm": AQLMConfig,
     "awq": AWQConfig,
     "deepspeedfp": DeepSpeedFPConfig,
-    "fp8": Fp8Config,
+    "fp8": Fp8Config if not is_hip() else Fp8FnuzConfig,  # type: ignore
```

Comment:
I would prefer for `fp8` to remain as `AutoFp8` and make `AutoFp8` integration compatible with AMD. E.g. in `fp8` we do things like quantizing in place. It would be nice if the AMD backend could take advantage of this

Could we use different quantization argument for this. E.g. `--quantization quark`